projects/project2/output/memorytest.out

```diff
@@ -0,0 +1,22 @@
+success
+success
+success
+success
+failure
+1 50
+failure
+success
+failure
+failure
+failure
+failure
+failure
+failure
+success
+failure
```

Comment:
<img width="1057" alt="image" src="https://user-images.githubusercontent.com/70302254/216448905-29a75a3d-d318-4772-89da-1d24c07f7db0.png">
I believe an allocated page to the 69 should allow reading since the read function does not have a specification that the addr has to be written to before. I get what you might be thinking since the old values in the address might be garbage from the last allocation such as in my input. I have made a piazza post about it. Will address it later after the answer

Reply:
I talked to Mike and he said he'll never read memory that hasn't been allocated. I'm going to fix this case and then should be good.